app/src/main/java/com/github/multimatum_team/multimatum/model/DeadlineAdapter.kt

```diff
@@ -16,21 +16,28 @@ Class which is used to show the deadline in a clear list.
 Based on the tutorial:
 https://www.raywenderlich.com/155-android-listview-tutorial-with-kotlin
  */
-class DeadlineAdapter(private val context: Context, private val dataSource: List<Deadline>) : BaseAdapter() {
+class DeadlineAdapter(private val context: Context) : BaseAdapter() {
     companion object {
         // value who define when there is not much time left for a deadline
         const val URGENT = 5
         const val PRESSING = 10
     }
 
+    private var dataSource: List<Deadline> = listOf()
+
     private val inflater: LayoutInflater =
         context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
 
+    fun submitList(deadlines: List<Deadline>) {
```

Comment:
I'm not really sure of what this function do, does it add deadlines or update them ?

Reply:
It updates the entire list. I took the name from a tutorial, but I agree it could have a less confusing name.